internal/datanode/data_node.go

```diff
@@ -1003,22 +1048,21 @@ func (node *DataNode) AddSegment(ctx context.Context, req *datapb.AddSegmentRequ
 			}, nil
 		}
 	}
-	// Update # of rows of the given segment.
-	ds.replica.updateStatistics(req.GetSegmentId(), req.GetRowNum())
+	ds.flushingSegCache.Remove(req.GetSegmentId())
```

Comment:
what is flushingSegCache? why we have to remove here?

Reply:
It is weird to remove something not added by me

Reply:
@congqixia any idea about why flushing seg cache is necessary?
I have the intuition that this part is way too complicated

Reply:
> @congqixia any idea about why flushing seg cache is necessary? I have the intuition that this part is way too complicated

`flushing seg cache` is used to prevent flushing same segment multiple times.